detailed_spec/requirements.tex

```diff
@@ -1,30 +1,38 @@
 \chapter{Requirements}
 
-\todo{Tohle by mozna nebylo spatny rovnou pojmenovat nejak jako `Features', `API' nebo mozna `Interfaces'.}
+This chapters outlines the main features that the project should implement to be a working product. All the features below have been discussed with professional HLASM developers and have been agreed on by the company.
 
--co ten nas produkt ma byt
-vseobecne zhrnutie
+The final product is a VS Code extension, downloadable from the Market Place. The extension contains all executables/libraries that are needed for the project to work correctly on the most popular platforms. No other prerequisites should be needed.
 
-...je to extension ... doda support pre ... 
--cela tato sekcia uz je popisana niekde na CA wiki, mozno dobry zaklad
+Modularity of the project is another important requirement. 
+
+The parsing library provides 2 kinds of API: complex one that mimicks the LSP specification and a simple one that accepts text and dependency-resolver object and returns diagnostics. 
+
+The language server implements the LSP standard, hence it can be easily reintegrated within other IDEs such as Eclipse Che.
 
 \section{Language features}
--zoznam veci jazyka co podporujeme
+This part provides a brief overview of the parts of HLASM that will be implemented in  the project.
+
+First of all, the project recognizes the syntax of HLASM (including continuations) and parses it into predefined structures. 
+
+It is able to interpret high-level parts of the assembler such as the conditional assembly instructions for the code generation, assembler instruction for the code layout determination and the macro expansion.
```

Comment:
tu by som si vedel predstavit vyslovene zoznam instrukcii ktore podporujeme a potom nice to have instrukcii?

Reply:
CA + assembler instrukcii je celkom dost... ozaj to tam chceme?

Reply:
da sa napisat ze podporujeme vsetky CA a asi 8 asm mame.. je to na tebe

Reply:
ja len ze tu mame sancu ukazat ake komplikovane je to co sme implementovali a kolko vela toho je